From 47ad209f6569a2cef661f6f3e4f2d2e9fb41693c Mon Sep 17 00:00:00 2001
From: Felix Kuehling <Felix.Kuehling@amd.com>
Date: Wed, 15 Nov 2017 18:01:11 -0500
Subject: [PATCH 2193/4131] drm/amdkfd: Fix debugfs dumping of KFD HQDs

Use shared_resources.queue_bitmap to determine the queues available
for KFD in each pipe.

Change-Id: I621bcd1fe9baa55084c231ff9d9bee46f176fd5c
Signed-off-by: Felix Kuehling <Felix.Kuehling@amd.com>
---
 drivers/gpu/drm/amd/amdkfd/kfd_device_queue_manager.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/gpu/drm/amd/amdkfd/kfd_device_queue_manager.c b/drivers/gpu/drm/amd/amdkfd/kfd_device_queue_manager.c
index bd1dcb2..e263378 100644
--- a/drivers/gpu/drm/amd/amdkfd/kfd_device_queue_manager.c
+++ b/drivers/gpu/drm/amd/amdkfd/kfd_device_queue_manager.c
@@ -1724,7 +1724,13 @@ int device_queue_manager_debugfs_hqds(struct seq_file *m, void *data)
 	int r = 0;
 
 	for (pipe = 0; pipe < get_pipes_per_mec(dqm); pipe++) {
+		int pipe_offset = pipe * get_queues_per_pipe(dqm);
+
 		for (queue = 0; queue < get_queues_per_pipe(dqm); queue++) {
+			if (!test_bit(pipe_offset + queue,
+				      dqm->dev->shared_resources.queue_bitmap))
+				continue;
+
 			r = dqm->dev->kfd2kgd->hqd_dump(
 				dqm->dev->kgd, pipe, queue, &dump, &n_regs);
 			if (r)
-- 
2.7.4

